pyquil/api/_qvm.py

```diff
@@ -173,19 +152,25 @@ def run(self) -> "QVM":
         )
         response = self._qvm_client.run_program(request)
         ram = {key: np.array(val) for key, val in response.results.items()}
-        self._memory_results.update(ram)
+        result_memory.update(ram)
 
-        return self
+        return QAMExecutionResult(executable=executable, memory=result_memory)
 
-    def augment_program_with_memory_values(self, quil_program: Program) -> Program:
-        p = Program()
+    def get_results(self, execute_response: QVMExecuteResponse) -> QAMExecutionResult:
+        """
+        Return the results of execution on the QVM.
 
-        for k, v in self._variables_shim.items():
-            p += MOVE(MemoryReference(name=k.name, offset=k.index), v)
+        Because QVM execution is synchronous, this is a no-op which returns its input.
+        """
+        return cast(QAMExecutionResult, execute_response)
```

Comment:
I don't think this cast is valid, as the objects have different shapes. Even if they were, I would recommend forming a `QAMExecutionResult` explicitly for clarity.

Reply:
Great catch!